### ISSUE TITLE
Codegen add with apiId is asking to delete when trying to push on new system

### ISSUE THREAD

**Issue by SebSchwartz** [aws-amplify/amplify-cli#3152]
**Describe the bug**
When adding codegen to a project (appsync not created with amplify) with apiId option. If a new developper (or ci/cd in my case) init the amplify project selecting an existing env, amplify status will tells him that pushed changes will delete it.

**Amplify CLI Version**
4.11.0

**To Reproduce**

- Create an amplify project and add codegen: amplify add codegen --apiId AppSyncAPIid
- Push changes and push in git also (.graphqlconfig.yml is also in git repo)
- Checkout project on another system (ci/cd codebuild)
- Execute amplify init to select right role,...
- amplify status will show:
| Category | Resource name | Operation | Provider plugin |
| -------- | ----------------------- | --------- | ----------------- |
| Auth | xxx | No Change | awscloudformation |
| Hosting | xx | No Change | awscloudformation |
| Api | xxx | Delete | |
| Function | xx | No Change | awscloudformation |

Only way to get normal status again is to remove api and add it again... BUT I got this problem inside a CICD so I cannot walkthrough the steps to add the API... And it doesn't seem to have an headless version of codegen add from what I saw from the sources.

 **Expected behavior**
All category should be in "No Change" state and the ci/cd should get the amplify working fine after init.

**Comment by SebSchwartz** [comment]
@UnleashedMind Any news on that?  

Thanks,
Sebastien

**Comment by SebSchwartz** [comment]
@UnleashedMind @kaustavghosh06  Still not ok and blocking CI/CD :(

**Comment by UnleashedMind** [comment]
@SebSchwartz 
I just tested and  the behavior is different now.

After going through all the steps, the amplify status will not show the api resources, so, not delete status for it. 
The generated mutations/queries are already in your frontend project, so your frontend project should work as normal. 

If your AppSync project is updated, you just need to run the `amplify add codegen --apiId AppSyncAPIid` command again to get the updated generated code.

Closing the issue for now, feel free to an open new issue if needed.
